### PR TITLE
Feature/batch timetable generation

### PR DESCRIPTION
**Pull request by Joshua27**
Since we decided not to block other Tasks while the Batch Generation is running it is now valid to start another task during the generation of all timetables. All other requirements according to #7 should work as desired.

### CODE REVIEW

File: src/main/java/de/hhu/stups/plues/ui/components/BatchResultBoxFactory.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+import de.hhu.stups.plues.data.entities.Course;
+import de.hhu.stups.plues.tasks.PdfRenderingTask;
+import javafx.scene.layout.VBox;
+
+import java.nio.file.Path;
+import java.util.Set;
+
+public interface BatchResultBoxFactory {
+  BatchResultBox create(@Assisted("major") Course major, @Assisted("minor") Course minor,
```
Comment (bivab) [inline src/main/java/de/hhu/stups/plues/ui/components/BatchResultBoxFactory.java:12]: Only the named assisted arguments (major, minor) need the annotation

File: src/main/java/de/hhu/stups/plues/ui/controller/BatchTimetableGeneration.java [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+    this.boxPool = new HashSet<>();
+
+    try {
+      this.tempDirectoryPath = Files.createTempDirectory("plues_timetables");
+    } catch (final IOException exception) {
+      logger.log(Level.INFO, "Could not create temporary directory.");
+    }
+
+    // delete the temporary directory when the application closes
+    Runtime.getRuntime().addShutdownHook(new Thread(() -> {
```
Comment (bivab) [inline src/main/java/de/hhu/stups/plues/ui/controller/BatchTimetableGeneration.java]: As it is a tmp directory I think we can just ignore it after the application closes

Comment (elrippoblue) [reply src/main/java/de/hhu/stups/plues/ui/controller/BatchTimetableGeneration.java]: This means deleting the lines 111 and following?

Comment (bivab) [reply src/main/java/de/hhu/stups/plues/ui/controller/BatchTimetableGeneration.java]: Yes

File: src/main/java/de/hhu/stups/plues/ui/components/BatchResultBox.java [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+  @Inject
+  public BatchResultBox(final FXMLLoader loader,
+                        final PdfRenderingTaskFactory taskFactory,
+                        @Assisted("major") final Course major,
+                        @Nullable @Assisted("minor") final Course minor,
+                        @Assisted final Path tempDirectoryPath,
+                        @Assisted final Set<PdfRenderingTask> taskPool) {
+    super();
+    this.task = taskFactory.create(major, minor);
+    this.majorCourse = new SimpleObjectProperty<>(major);
```
Comment (bivab) [inline src/main/java/de/hhu/stups/plues/ui/components/BatchResultBox.java]: I'm not sure these need to be properties

**Pull request merged**